src/sentry/sentry_metrics/consumers/indexer/processing.py

```diff
@@ -108,19 +108,19 @@ def _process_messages_impl(
         batch.filter_messages(cardinality_limiter_state.keys_to_remove)
 
         extracted_strings = batch.extract_strings()
-        org_strings = next(iter(extracted_strings.values())) if extracted_strings else {}
 
-        sdk.set_measurement("org_strings.len", len(org_strings))
+        sdk.set_measurement("extracted_strings.len", len(extracted_strings))
 
         with metrics.timer("metrics_consumer.bulk_record"), sentry_sdk.start_span(op="bulk_record"):
-            record_result = self._indexer.bulk_record(
-                use_case_id=self._config.use_case_id, org_strings=org_strings
-            )
+            record_result = self._indexer.bulk_record(extracted_strings)
 
         mapping = record_result.get_mapped_results()
         bulk_record_meta = record_result.get_fetch_metadata()
 
-        new_messages = batch.reconstruct_messages(mapping, bulk_record_meta)
+        new_messages = batch.reconstruct_messages(
+            next(iter(mapping.values())) if mapping else {},
+            next(iter(bulk_record_meta.values())) if bulk_record_meta else {},
+        )
```

Comment:
We are doing this because `batch.reconstruct_messages` is not use case aware yet

Reply:
The plan is to do that in this PR, right?

Reply:
why is this resolved?

Reply:
I think if `bulk_record` can accept multiple use-cases, then `reconstruct_messages` should be able to as well. Does this potentially leave us in an incorrect state if somebody passed a batch with two use_case_id's to the indexer consumer?

I'm not saying we have to support batches with two different use_case_id's immediately but I don't think it's a good idea to let the data pass through without throwing up some kind of error if we can't process it. Maybe we can assert somewhere that there is only one use_case_id per batch, and remove that assertion later.

Reply:
I updated reconstruct_messages to be use case aware (i.e. uses the `use_case_id` from the intermediate type to obtain the `str` -> `int` mapping)

Reply:
> why is this resolved?

It was resolved because I had a discussion with Riya in slack